python/pycrdt/text.py

```diff
@@ -103,19 +103,30 @@ def __setitem__(self, key: int | slice, value: str) -> None:
                 del self[key]
                 self.integrated.insert(txn._txn, key, value)
             elif isinstance(key, slice):
-                start, stop = self._check_slice(key)
-                if start != stop:
-                    self.integrated.remove_range(txn._txn, start, stop)
+                start, length = self._check_slice(key)
+                if length > 0:
+                    self.integrated.remove_range(txn._txn, start, length)
                 self.integrated.insert(txn._txn, start, value)
             else:
                 raise RuntimeError(f"Index not supported: {key}")
 
+    def replace_range(self, value: str, start: int, stop: int) -> None:
+        """Replace the range of characters in the interval ['start', 'stop') with 'value'."""
+        del self[start:stop]
+        self[start:start] = value
+
+    def remove_range(self, start: int, stop: int) -> None:
+        """Remove the range of characters in the interval ['start', 'stop')."""
+        del self[start:stop]
+
+    def insert(self, value: str, index: int) -> None:
```

Comment:
For me, reading this like "insert 'value' at index 'index'" feels just a bit more natural than "insert at index 'index' 'value'". I also wanted it to be similar to my suggested `replace_range`, which would have the same signature as [`replace` from Textual's `TextArea` widget](https://github.com/Textualize/textual/blob/9f43d300052f3a2ef8490a522e41463897477136/src/textual/widgets/_text_area.py#L1627). I thought: Since `insert` and `clear` are somewhat verbose convenience functions, I make them a bit more convenient for me.
However, I realized just now that the current implementation follows the signature also used in `list.insert` as well as in the underlying Yrs `TextRef` `insert` method.

Swapping arguments here would be inconsistent and thus might be more confusing than helpful.

What do you think?